File: api_server/views/System.py
from flask import Blueprint
from controller.UserManager import UserManager
from middleware import require_argument, to_json, verify_token
from tokenProcessor import TokenProcessor
from config import refresh_salt
from platform_info import RoseliaSysInfo

token_processor = TokenProcessor()

system_view = Blueprint('system_view', __name__, url_prefix='/api/system')

route = system_view.route


@route('/refresh-salt', methods=['POST'])
@to_json
@verify_token(1, True)
def refresh(username, role):
    success = False
    if UserManager.is_master(username):
        success = True
        refresh_salt()

    return {
        'success': success
    }


@route('/basic-info')
@to_json
@verify_token(1)
def get_basic_info(username, role):
    if UserManager.is_master(username):
        return {
            'success': True,
            'result': {
                'os': RoseliaSysInfo.os_full_name(),
                'cpu': RoseliaSysInfo.processor_info()
            }
        }
    return {
        'success': False,
        'msg': 'Not master'
    }


@route('/dynamic-info')
@to_json
@verify_token(1)
def get_dynamic_info(username, role):
    if UserManager.is_master(username):
        return {
            'success': True,
            'result': {
                'cpu': RoseliaSysInfo.cpu_usage(),
                'memory': RoseliaSysInfo.memory_usage()
            }
        }

File: RoseliaBlog/Controllers/System.cs
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using RoseliaBlog.Middleware;
using RoseliaBlog.Models;
using RoseliaBlog.RoseliaCore;
using RoseliaBlog.RoseliaCore.Token;

namespace RoseliaBlog.Controllers
{
    [Route("/api/system")]
    public class SystemController : Controller
    {
        [Route("refresh-salt")]
        [MasterUserAuthentication(RequiredTokenType = TokenTypes.RoseliaTokenType.UserCredential)]
        public IActionResult RefreshSalt()
        {
            RoseliaCore.Config.RefreshSalt();
            return RoseliaApiResult.Succeed("").ToActionResult();
        }

        [Route("basic-info")]
        [MasterUserAuthentication(RequiredTokenType = TokenTypes.RoseliaTokenType.UserCredential)]
        public IActionResult GetBasicSystemInfo()
        {
            return RoseliaApiResult.Succeed(new
            {
                Os = PlatformInfo.CurrentPlatformInfo.Value,
                Cpu = PlatformInfo.CurrentCpuInfo.Value
            }).ToActionResult();
        }

        [Route("dynamic-info")]
        [MasterUserAuthentication(RequiredTokenType = TokenTypes.RoseliaTokenType.UserCredential)]
        public IActionResult GetDynamicSystemInfo()
        {
            return RoseliaApiResult.Succeed(new
            {
                Cpu = PlatformInfo.CpuUsage.GetUsage,
                Memory = PlatformInfo.MemoryUsage.GetUsage
            }).ToActionResult();
        }
    }
}
